server/store/kvstore/upgrade.go

```diff
@@ -62,3 +71,48 @@ func (s *Store) shouldPerformUpgrade(currentSchemaVersion, expectedSchemaVersion
 	}
 	return false
 }
+
+func upgradeTo14(s *Store) error {
+	var allKeys []string
+	i := 0
+	for {
+		keys, appErr := s.api.KVList(i, perPage)
+		if appErr != nil {
+			return errors.Wrap(appErr, "failed to list poll keys")
+		}
+
+		if len(keys) == 0 {
```

Comment:
```suggestion
		if len(keys) < perPage {
```

This would reduce the number of invoking `KVList` by one in many cases

Reply:
Good point :+1: Updated